static/js/components/Editor.js

```diff
@@ -0,0 +1,310 @@
+      icon:    <i className="material-icons format_italic">format_italic</i>,
+      mark:    schema.marks.em
+    }
+  ],
+  [
+    {
+      command: wrapInList(schema.nodes.bullet_list),
+      title:   "bullet list",
+      icon:    (
+        <i className="material-icons format_list_bulleted">
+          format_list_bulleted
+        </i>
+      )
+    },
+    {
+      command: wrapInList(schema.nodes.ordered_list),
+      title:   "ordered list",
+      icon:    (
+        <i className="material-icons format_list_numbered">
+          format_list_numbered
+        </i>
+      )
+    },
+    {
+      command: wrapIn(schema.nodes.blockquote),
+      title:   "quote",
+      icon:    <i className="material-icons format_quote">format_quote</i>
+    }
+  ]
+]
+
+export const newLinkForm = () => ({
+  url: ""
+})
+
+type Props = {
+  onChange: (serialized: string) => void,
+  initialValue?: string,
+  autoFocus?: boolean,
+  dispatch: Dispatch<*>
+}
+
+type State = {
+  editorState: Object,
+  showLinkMenu: boolean
+}
+
+export class Editor extends React.Component<Props, State> {
+  node: { current: null | React$ElementRef<typeof HTMLDivElement> }
+  view: Object
+  uuid: string
+  formHelpers: Object
+  dispatchTransaction: Function
+
+  constructor(props: Props) {
+    super(props)
+    this.node = React.createRef()
+    this.uuid = makeUUID(10)
+
+    const { getForm, actionCreators } = configureForm(this.uuid, newLinkForm)
+    this.formHelpers = {
+      getForm,
+      ...actionCreators
+    }
+
+    const { initialValue } = props
+    const editorStateOptions = {
+      doc:     defaultMarkdownParser.parse(initialValue || ""),
+      plugins: exampleSetup({
+        schema,
+        menuBar: false
+      })
+    }
+
+    this.state = {
+      editorState:  EditorState.create(editorStateOptions),
+      showLinkMenu: false
+    }
+    this.dispatchTransaction = this.dispatchTransaction.bind(this)
+  }
+
+  componentDidMount() {
+    const { autoFocus } = this.props
+    const { editorState } = this.state
+
+    if (this.node.current) {
+      this.view = new EditorView(this.node.current, {
+        state:               editorState,
+        dispatchTransaction: this.dispatchTransaction
+      })
+
+      if (autoFocus) {
+        this.view.focus()
+      }
+    }
+  }
+
+  dispatchTransaction(transaction: Object) {
+    // we need our own handler for this so we can hook into
+    // the EditorState lifecycle to synchronize it with the
+    // redux store (via the onChange prop)
+    const { onChange } = this.props
+
+    const state = this.view.state.apply(transaction)
+    this.view.updateState(state)
+    this.setState({ editorState: state })
+
+    onChange(defaultMarkdownSerializer.serialize(this.view.state.doc))
+  }
+
+  menuClickHandlerCreator = (command: Function) => (e: Object) => {
+    e.preventDefault()
+    command(this.view.state, this.view.dispatch, this.view)
+    this.view.focus()
+  }
+
+  renderMenuBar = () => {
```

Comment:
Personally I like keeping the rendering methods next to each other. I'd prefer it if this function was defined right above the main `render` method but I'll leave that to you